src/main/java/org/smartregister/pathevaluator/action/ActionHelper.java

```diff
@@ -82,8 +82,9 @@ public static ResourceType getResourceType(Action action) {
 				return clientDao.findFamilyMemberyByJurisdiction(jurisdiction.getCode());
 
 			case TASK:
-			case GLOBAL_TASK:
 				return taskDao.findTasksByJurisdiction(jurisdiction.getCode(),planIdentifier);
+			case GLOBAL_TASK:
+				return taskDao.findTasksByJurisdiction(jurisdiction.getCode());
```

Comment:
What's the use case of not passing plan-identifier here?
I think we need to identify the use case of why it was actually implemented to filter out using plans.

Reply:
It seems a big difficult to pass null since all the methods calling this method pass the following `planDefinition.getIdentifier()` .  

- This is the PR that added the changes https://github.com/OpenSRP/opensrp-plan-evaluator/pull/108
- I think it might be a bug based on this issue https://github.com/OpenSRP/opensrp-plan-evaluator/issues/63

Here

![Screenshot from 2021-03-17 18-32-52](https://user-images.githubusercontent.com/31766075/111612590-ab400d80-87ee-11eb-99fb-2a5468587849.png)

Reply:
Ya looks good. I am a bit uncertain about the change we are trying to make! Since, it might impact other Global Tasks evaluation, as we are removing the filtration on the basis of a plan. 
Can't we pass `planDefinition.getIdentifier()` as is? And handle it on the server core, add it in the criteria for filtration if found as not null/empty.

One more thing is it possible to identify a criteria to distinguish between the current use case and the older ones? In that case, we can easily strip away the identifier and ensure backward compatibility as well. 
What are your thoughts on this?

Reply:
I think we can, I am not sure of the impact of stripping of the identifier from the actual plan passed to achieve this. 
The plan is passed as a complete object, so I don't know the impact of stripping away the identifier will have on other methods/functionality depending on this. 

> One more thing is it possible to identify a criteria to distinguish between the current use case and the older ones? In that case, we can easily strip away the identifier and ensure backward compatibility as well.
What are your thoughts on this?

:thinking:  I think the only way would be to hard code the Goldsmith plan IDs since the actual scheduling/action trigger code is on client core

I agree that we should support prior implementations but I'm not sure how to go about that.

@Rkareko What is the impact/use of Global tasks currently on Reveal?

Reply:
@ekigamba Global tasks are used to keep state between plans e.g if a family is registered on Plan A for structure A, then on plan B the same structure A should show that a family has been registered.

Have a look at this for context on why the plan identifier is passed as a param
![global_tasks](https://user-images.githubusercontent.com/47570855/112122129-0304bd00-8bd1-11eb-8598-0d52d660a25e.png)